_summaries/bello2021lambdanetworks_1.md

```diff
@@ -0,0 +1,42 @@
+---
+layout: summary
+title: Summary
+paper: {{hendrycks2016gaussian_1}}
+# Please fill out info below
+author:  marcobueso
+score:  10
+---
+
+
+* What is the core idea?\
+Authors introduce Lambda Networks, which are composed of lambda layers.\
+A lambda layer is a class of layer that models content and position based interactions, without using expensive memory maps.\
+Figure 1 displays a lambda layer vs. a self-attention map. It is clear lambda layers are much cheaper computationally.\
+![Bello (2021).](bello_lambda_networks_2021_1_a.PNG)
+In the words of the author: "Whereas self-attention defines a similarity kernel between the query and the context elements, a lambda layer instead summarizes contextual information into a fixed-size linear function". The fact that it becomes a linear function is what makes is less expensive.\
+At the first line, I mention position-based interactions. These are interactions which take into account the relative distance between between the context and the query (e.g. a pixel and surrounding pixels).
+
+* How is it realized (technically)?\
+Generally, we have an query input $$(q_n, n)$$, which gets mapped to an output vector $$y_n$$ given the context $$C$$ as:\
+$$((q_n, n),C)\leftarrow y_n$$ \
+The problem here is that this function ignores the position based interaction $$(q_n, (n,m))$$. Lambda layers take into account these relation (n, m) between the query and context element positions.\
+Figure 2 shows the computational graph of the lambda layer.\
+![Bello (2021).](bello_lambda_networks_2021_1_b.PNG)
+The author also provides pseudocode to create a lambda layer.
+
+* How well does the paper perform?\
+The LambdaNetworks were tested against self-attention models, and resulted in better performance, as well as being more computationally efficient, and faster than the other models, which were sometimes slowed down and hindered by large memory costs, impractical implementations, or the requirement of huge amounts of data.\
+The most impressive results are:\
+LambdaResNets achieve a 4.4x speedup over EfficientNets on ImageNet, and they are still more energy efficient.\
+In a semi-supervised learning environment, LambdaResNets achieve up to a 9.5x speedup over EfficientNet NoisyStudent checkpoints (a kind of semi-supervised learning approach).
+
+
+* What interesting variants are explored?\
+The authors explore wiht hybrid architectures. They do this because even with the time and memory saving compared to self-attention, the lambda layer still incurs a quadratic time complexity. The authors study a convolution-lambda hybrid, which increases representational power at no throughput cost.
```

Comment:
I believe they also explore a larger model and see how well it does. Might be an interesting add here.